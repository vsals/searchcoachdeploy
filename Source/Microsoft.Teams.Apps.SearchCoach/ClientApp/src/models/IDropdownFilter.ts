// <copyright file="IDropdownFilter.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/**
* Interface to work with drop-down list item values.
*/
export interface IDropdownListItem {
    name: string;
    id: string;
}

/**
* Interface to work with drop-down list selected item value.
*/
export interface ISelectedDropdownItem {
    header: string,
    key: string,
}