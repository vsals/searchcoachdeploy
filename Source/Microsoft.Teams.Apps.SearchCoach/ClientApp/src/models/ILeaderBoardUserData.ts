// <copyright file="ILeaderBoardUserData.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// Interface to deal with user response data for leader-board tab.
export interface ILeaderBoardUserData {
    userName: string;
    rightAnswers: number;
    questionsAttempted: number;
}