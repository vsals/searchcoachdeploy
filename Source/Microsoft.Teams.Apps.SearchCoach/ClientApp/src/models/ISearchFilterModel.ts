// <copyright file="ISearchFilterModel.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/** This interface is used for maintaining the state of domain check-box values. */
export interface ISearchFilterModel {
    // searched text by user.
    searchText: string;
    // selected domains using domain filter
    domains?: Array<string>;
    // selected freshness value using freshness filter
    freshness: string;
    // selected country-code using location filter
    market: string;
}