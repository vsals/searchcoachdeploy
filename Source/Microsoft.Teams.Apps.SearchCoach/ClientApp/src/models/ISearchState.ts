// <copyright file="ISearchState.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

/**
* Search state interface.
*/
export interface ISearchState {
    // search text string entered by user.
    searchText: string;
    // country selected string by user.
    selectedCountry: string;
}