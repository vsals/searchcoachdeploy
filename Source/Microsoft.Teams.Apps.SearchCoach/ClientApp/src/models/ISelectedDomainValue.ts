// <copyright file="ISelectedDomainValue.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// This interface is used for maintaining the state of domain check-box values.
export interface ISelectedDomainValue {
    // unique value to identity each item.
    key: string;
    // name of the check-box item.
    title: string;
    // explains whether the check-box is checked or unchecked.
    isChecked: boolean;
}