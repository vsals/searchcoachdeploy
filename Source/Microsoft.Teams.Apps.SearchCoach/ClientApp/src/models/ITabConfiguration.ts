// <copyright file="ITabConfiguration.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

// Interface to deal with mapping of teams configurable tab.
export interface ITabConfiguration {
    tabId?: string,
    teamId: string,
    groupId: string,
}