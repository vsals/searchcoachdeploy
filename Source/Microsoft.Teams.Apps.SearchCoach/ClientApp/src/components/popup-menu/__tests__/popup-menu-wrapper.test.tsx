// <copyright file="popup-menu-wrapper.test.tsx" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import * as React from "react";
import { Provider } from "@fluentui/react-northstar";
import PopUpMenuWrapper from "../popup-menu-wrapper";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { ISelectedDropdownItem } from "../../../models/IDropdownFilter";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: any) => key,
        i18n: { changeLanguage: jest.fn() },
    }),

    withTranslation: () => (Component: any) => {
        Component.defaultProps = {
            ...Component.defaultProps,
            t: (key: any) => key,
        };
        return Component;
    },
}));

jest.mock("@microsoft/teams-js", () => ({
    initialize: () => {
        return true;
    },
    getContext: (callback: any) =>
        callback(
            Promise.resolve({ teamId: "ewe", entityId: "sdsd", locale: "en-US" })
        ),
}));

let container: any = null;

beforeEach(() => {
    // setup a DOM element as a render target.
    container = document.createElement("div");
    // container *must* be attached to document so events work correctly.
    document.body.appendChild(container);
});

describe("PopUpWrapperContent", () => {
    it("renders snapshots", () => {
        act(() => {
            render(
                <Provider>
                    <PopUpMenuWrapper
                        activeButton="operators"
                        onCountryChange={(value: ISelectedDropdownItem) => { }}
                        onDomainValuesChange={(selectedDomains: string) => { }}
                        onFreshnessChange={(value: any) => { }}
                        onOperatorAdded={(value: string) => { }}
                        selectedCountry={{ header: "No Filter", key: "nf" }}
                        selectedDomains=""
                    />
                </Provider>,
                container
            );
        });

        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});