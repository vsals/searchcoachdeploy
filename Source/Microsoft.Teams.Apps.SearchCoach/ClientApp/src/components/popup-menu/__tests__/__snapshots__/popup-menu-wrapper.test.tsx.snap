// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpWrapperContent renders snapshots 1`] = `"<div class=\\"ui-provider\\" dir=\\"ltr\\" data-uses-unhanded-props=\\"true\\"><button aria-disabled=\\"false\\" data-aa-class=\\"PopupButton\\" class=\\"ui-button\\" aria-haspopup=\\"dialog\\" data-uses-unhanded-props=\\"true\\"></button></div>"`;
